src/views/signUpBuyer.jsx

```diff
@@ -0,0 +1,136 @@
+import React, { useState } from 'react';
+import { NavLink, useNavigate } from 'react-router-dom';
+import { ToastContainer, toast } from 'react-toastify';
+import CART from '../assets/CART.png';
+
+function SignUpBuyer() {
+  const navigate = useNavigate();
+  const [btn, setbtn] = useState({
+    display: 'hidden',
+    text: `Sign Up`,
+  });
+
+  const allFields = [
+    { placeh: 'enter first name ', label: ' first name', type: 'text' },
+    { placeh: 'enter last name ', label: ' last name', type: 'text' },
+    { placeh: 'enter email ', label: ' email', type: 'email' },
+    { placeh: 'phone ', label: ' phone', type: 'tel' },
+    { placeh: 'password ', label: ' password', type: 'password' },
+    {
+      placeh: 'confirm password ',
+      label: 'confirm password',
+      type: 'password',
+    },
+  ];
+  const SignUp = async (e) => {
+    e.preventDefault();
+    setbtn({
+      display: 'inline-block',
+      text: `Loading...`,
+    });
+    const phoneNumberRegex = /[a-zA-Z]/;
+    if (phoneNumberRegex.test(e.target[3].value) == true) {
+      toast.error('incorrect phone number', { theme: 'colored' });
+      e.target[3].focus();
+    } else {
+      if (e.target[4].value == e.target[5].value) {
+        await fetch('http://localhost:4000/api/v1/register', {
+          method: 'POST',
+          mode: 'cors',
+          headers: {
+            'Content-Type': 'application/json',
+          },
+          body: JSON.stringify({
+            firstname: e.target[0].value,
+            lastname: e.target[1].value,
+            email: e.target[2].value,
+            phone: e.target[3].value,
+            password: e.target[4].value,
+          }),
+        })
+          .then((res) => res.json())
+          .then((res) => {
+            console.log(res);
+            if (res.status[0] == 's') {
+              toast.success(res.message, { theme: 'colored' });
+              setTimeout(() => {
+                navigate('/login');
+              }, 2000);
+            } else {
+              toast.error(res.message, { theme: 'colored' });
+            }
+          });
+      } else {
+        e.target[5].focus();
+        toast.error('confirm the correct password', { theme: 'colored' });
+      }
+    }
+    setbtn({
+      display: 'hidden',
+      text: `Sign Up`,
+    });
+  };
+
+  return (
+    <div className="bg-[#D9D9D9]  min-h-[100vh] pt-20 ">
+      <ToastContainer />
```

Comment:
Please render the <ToastContainer /> component at a high level component like App or index to allow all components to use it without having to import and render it individually in each component. This can help improve the consistency and maintainability of the application, as well as reduce the amount of code duplication.

Reply:
fixed